wait-for-deployment

```diff
@@ -36,28 +37,48 @@ get_deployment_jsonpath() {
   kubectl get deployment "${deployment}" -o "jsonpath=${_jsonpath}"
 }
 
+get_statuses_for_deployment() {
+  kubectl get pods --selector=app=${deployment} -o 'jsonpath={.items[*].status.conditions[*].type}'
+}
+
+count_ready_pods() {
+  get_statuses_for_deployment | tr ' ' '\n' | grep Ready | wc -l
+}
+
 if [[ $# != 1 ]]; then
-  echo "usage: $(basename $0) <deployment>" >&2
+  echo "Usage: $(basename $0) <deployment>" >&2
   exit 1
 fi
 
 readonly deployment="$1"
 
 readonly generation=$(get_generation)
-echo "waiting for specified generation ${generation} to be observed"
-while [[ $(get_observed_generation) -lt ${generation} ]]; do
+echo "Waiting for specified generation: ${generation} for deployment ${deployment} to be observed"
+
+current_generation=$(get_observed_generation)
+while [[ ${current_generation} -lt ${generation} ]]; do
   sleep .5
+  echo "Current generation: ${current_generation}, expected generation: ${generation}, waiting"
+  current_generation=$(get_observed_generation)
 done
-echo "specified generation observed."
+echo "Observed expected generation: ${generation}"
```

Comment:
`s/${generation}/${current_generation}/` since we use the less-than (`-lt`) operator above now.